src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2377,6 +2377,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 {
                     Error(diagnostics, ErrorCode.ERR_RefAssignmentMustHaveIdentityConversion, expression.Syntax, targetType);
                 }
+                else
+                {
+                    return expression;
+                }
```

Comment:
I see, that makes sense I guess. There is still one thing concerns me specifically for tuples. It looks like warnings reported by TupleTypeSymbol.ReportNamesMismatchesIfAny will not be reported in this case, but they will be for ByValue assignments and returns. Did you, guys, make an explicit decision about this behavior?